File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/fight/GameActionFightStealKamaMessage.scala





















// Generated on 05/03/2013 10:34:34
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameActionFightStealKamaMessage extends AbstractGameActionMessage {
  override val messageId = 5535

  protected trait Message extends AbstractGameActionMessage.Message with Serializable {
    override val messageId = GameActionFightStealKamaMessage.messageId

    val targetId: Int
    val amount: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(targetId)
      buf.writeShort(amount)
    }

  }

  protected def create(actionId: Short, sourceId: Int, targetId: Int, amount: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val sourceId = buf.readInt()
    val targetId = buf.readInt()
    val amount = buf.readShort()

    create(actionId, sourceId, targetId, amount)
  }

}

object GameActionFightStealKamaMessage extends GameActionFightStealKamaMessage {
  case class MessageImpl(actionId: Short, sourceId: Int, targetId: Int, amount: Short) extends Message
  type Target = MessageImpl

  override def create(actionId: Short, sourceId: Int, targetId: Int, amount: Short) = MessageImpl(actionId, sourceId, targetId, amount)

  def apply(actionId: Short, sourceId: Int, targetId: Int, amount: Short) = create(actionId, sourceId, targetId, amount)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameActionFightStealKamaMessage.java


// Generated on 05/08/2013 19:37:40
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionFightStealKamaMessage extends AbstractGameActionMessage {
    public static final int MESSAGE_ID = 5535;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int targetId;
    public short amount;
    
    public GameActionFightStealKamaMessage() { }
    
    public GameActionFightStealKamaMessage(short actionId, int sourceId, int targetId, short amount) {
        super(actionId, sourceId);
        this.targetId = targetId;
        this.amount = amount;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(targetId);
        buf.writeShort(amount);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        targetId = buf.readInt();
        amount = buf.readShort();
        if (amount < 0)
            throw new RuntimeException("Forbidden value on amount = " + amount + ", it doesn't respect the following condition : amount < 0");
    }
    
}
